vllm/worker/vineyard_llm_cache.py

```diff
@@ -388,40 +458,50 @@ def update_seq_kv_caches(
         for j in range(self.layer):
             self.cuda_buffer[:, j, :update_token_size].copy_(
                 kv_caches[j][:, slot_mapping // block_size, slot_mapping % block_size])
-        self.buffer.copy_(self.cuda_buffer)
+        update_buffer.copy_(self.cuda_buffer)
         end_unload.record()
         end_unload.synchronize()
         duration = start_unload.elapsed_time(end_unload) / 1000.0
         self.metrics.time_unload.append(duration)   
         self.metrics.normalized_time_unload.append(0 if update_token_size == 0 else duration/update_token_size)
-        
+
         start_time = time.perf_counter()
-        # updates into vineyard
-        updated = self.cache.update(
+
+        # async update
+        update_task = partial(self._update_kv_cache,
             prefix=update_prefix,
             tokens=update_tokens,
-            kv_cache_list=self.tensors[:update_token_size],
+            kv_cache_list=update_tensors[:update_token_size],
+            buffer_tensors_tuple=buffer_tensors_tuple,
         )
+        try:
+            logger.debug(
+                f"submit update task: #prefix={len(update_prefix)}, #tokens={len(update_tokens)}"
+            )
+            self._update_tasks.put_nowait(update_task)
+            logger.debug(
+                f"task queue size={self._update_tasks.qsize()}, tensor pool size={self.tensor_pool.size}"
+            )
+        except Full:
```

Comment:
Since the maximum number of update tasks are bounded by the size of `tensor_pool`, we should never overflow the update task queue, so we should never block here. If we encounter this exception, that means our flow-control mechanism has failed, so we should probably increment an error counter here so we can be alerted to this condition.

Reply:
added